### DISCUSSION TITLE
Yarn version detection in monorepo is wrong

### DISCUSSION THREAD

**Discussion by SimenB** [renovatebot/renovate#24955] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

We've set up autorefreshing of lockfiles in the Jest repo. Renovate then discovers a bunch of lockfiles in our e2e tests which it also wants to refresh (this is great 👍). But it's unable to discover in the nested directories that we're on Yarn v3, and tries to pass `--ignore-engines` when updating them.

See https://github.com/jestjs/jest/pull/14537#issuecomment-1722608964

Link to config: https://github.com/jestjs/jest/blob/46285d8bd27e1fd798903dc1b46e4b095af1a959/.github/renovate.json#L4-L6

### Relevant debug logs

A job log - not sure which part is relevant, sorry 😬 

https://developer.mend.io/github/jestjs/jest/-/job/c0858558-058f-4661-a2d6-ce185f5402f1

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
@SimenB can you tick the rebase/retry checkbox in https://github.com/jestjs/jest/pull/14537 so that we can get a more recent log?

**Reply by SimenB** [reply]
I did right before opening this - https://developer.mend.io/github/jestjs/jest/-/job/c0858558-058f-4661-a2d6-ce185f5402f1 is just over 3 hours old 🙂

**Reply by rarkins** [reply]
For some reason it's not running Yarn at all. I got the same when I tried forking it and running locally.

```
DEBUG: Writing any updated package files (branch="renovate/lock-file-maintenance")
DEBUG: No updated lock files in branch (branch="renovate/lock-file-maintenance")
```

**Reply by SimenB** [reply]
Huh, weird. Not sure what changed in the couple of weeks since that PR was opened?

Semi-related btw - automerging ignores its own advice to not merge the PR

https://github.com/jestjs/jest/pull/14508

**Reply by rarkins** [reply]
The newest commit didn't have any artifacts error. The warning applied to the previous commit

**Reply by rarkins** [reply]
And presumably you have tests set up which passed?

**Reply by SimenB** [reply]
Tests (or, typecheck) failed 
<img width="499" alt="image" src="https://github.com/renovatebot/renovate/assets/1404810/3c1bc26e-1160-46dd-b083-cf817d21a8ad">

But at the time I did not have branch protections set up. Maybe that's needed (in the same way that's needed if you want to use "auto merge" in GH)?

---

But that's orthogonal to this issue 😅 

Not sure why it's not failing anymore - it's not updating the lockfiles either. If I manually delete and recreate the lockfiles, almost all of them have a diff

```
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   e2e/async-regenerator/yarn.lock
        modified:   e2e/babel-plugin-jest-hoist/yarn.lock
        modified:   e2e/chai-assertion-library-errors/yarn.lock
        modified:   e2e/coverage-transform-instrumented/yarn.lock
        modified:   e2e/global-setup/yarn.lock
        modified:   e2e/global-teardown/yarn.lock
        modified:   e2e/native-esm/yarn.lock
        modified:   e2e/transform/multiple-transformers/yarn.lock
        modified:   e2e/transform/transform-snapshotResolver/yarn.lock
        modified:   e2e/transform/transform-testrunner/yarn.lock
        modified:   e2e/transform/transformer-config/yarn.lock
```

**Reply by rarkins** [reply]
Yes, it's not even trying to run Yarn. The problem is that the repo is so big it makes it every challenging to debug

**Reply by SimenB** [reply]
Yeah, understandable 😅

**Reply by rarkins** [reply]
I'll still have a try though.

And yes you're right - if using GitHub automerge _definitely_ have branch protections
